### PR TITLE
Feature budget binderv1

### PR DESCRIPTION
**Pull request by IsmaelMastronardi**
### Budget Binder 
Budget Binder is a Ruby On Rails application designed to keep track of transactions. 

### Functionality :
- Allows the user to create an account or log in with an already created account.
- The user can create new categories to organize their transactions.
- The user can create new transactions and add them to one or more groups.
- The user can see all their groups and the transactions contained in them.

### Pages : 
The Application contains the following pages : 

- Home page :
Screenshot (125)

- Categories page :
Screenshot (126)

- Entities page :
Screenshot (124)

- Create a Group page :
Screenshot (120)

- Create Entity page :
Screenshot (122)

### Video:
https://www.loom.com/share/d9b68a3ddfec49aea3180ee341c34153

### Render Link :
https://budget-binder.onrender.com/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Feel free to check the [issues page](https://github.com/IsmaelMastronardi/budget-app/issues).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I want to thank my family for their support and encouragement. Your love means the world to me.
```
Comment (Shedrack-Sunday) [inline README.md:161]: - [ ] The Original design idea is by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella). Kindly reference him in this section of the README as it is in the project requirements.

- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author.

File: app/views/entities/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+
+  <div id="entities" class="pt-16 pb-12 mx-4 mb-20 bg-light_grey">
+    <%= render @group.entities %>
+  </div>
+
+  <div class="w-full flex flex-row text-center justify-center items-center bg-green py-2 fixed bottom-0 left-0 right-0">
+    <%= link_to "New Transaction", new_user_group_entity_path(current_user, @group), 
+    class: "py-3 px-5 bg-blue-600 text-white block font-medium border py-4 px-20" %>
+  </div>
```
Comment (Shedrack-Sunday) [inline app/views/entities/index.html.erb]: - [ ] On the transaction page, the requirement is to have the sum of the total number of transaction of each category at the top of this transaction page. Kindly add it up 👏 

### Requirement
image

### View:
image

File: app/controllers/entities_controller.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  def edit; end
+
+  # POST /entities or /entities.json
+  def create
+    @entity = Entity.new(entity_params.except(:group_ids, :group_id))
+
+    entity_params[:group_ids].each do |group_id|
+      @entity.groups << current_user.groups.select { |group| group.id == group_id.to_i }
+    end
+
```
Comment (Shedrack-Sunday) [inline app/controllers/entities_controller.rb:29]: - [ ] Based on the project requirments, when  a user click (or taps) the "Save" button to create a new transaction(entity), It should re-direct  to the transactions page for that category. In your case, it leads back to category page. Kindly fix that  on the view 👍 

### Initial page in creation:
image

### Expected next navigation on save:
image

### Page it leads to instead:
image

## Requirements:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @IsmaelMastronardi      👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the rails testing features ✔️
- Kudus making personalizing the design      ✔️  

# - Required Changes  🌵  

- [ ] I made a couple of comments regarding some features you missed, mostly on paths when a user action is performed. Kindly look into this 👏 

- [ ] It will be nice to move the video and project link to the README as its a major path of the README, instead of placing it on the pull request 👍 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Install](#install)
+  - [Usage](#usage)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] As previously suggested by the reviewer, please attach the links to the presentation and live demo link in the README file for a professional repo. 🙏

File: app/views/entities/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<div class="w-full relative">
+  <div class="text-center bg-blue top-0 left-0 right-0 py-1 fixed">
+    <h1 class="font-bold text-4xl text-white"><%= @group.name %></h1>
+  </div>
+
+  <div class="text-center bg-blue mx-auto fixed top-10 left-0 right-0 p-1">
+    <p class="border-t-2 inline p-1">
+      Total Transactions: <%= @group.entities.size %>
+    </p>
+  </div>
```
Comment (MahmoudBakr23) [inline app/views/entities/index.html.erb]: - [ ] About the previous reviewer's suggestion for transactions' total amount is to display the total sum of the transactions' value not how many transactions are in the group. 🙏 
Let's say we have two transactions with $1000 value for each so the total amount of both would be $2000
BudgetApp - Google Chrome 11_24_2023 6_52_57 AM

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @IsmaelMastronardi 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but some have still not been addressed yet.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @IsmaelMastronardi     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the Awesome work 🥂  Keep Rocking 🥇 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README file  
- [x] Nice UI design 🥇 






 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**